Datastore/src/EntityMapper.php

```diff
@@ -204,22 +192,6 @@ public function convertValue($type, $value, $className = Entity::class)
                 break;
 
             case 'doubleValue':
-                // Flow will enter this logic only when REST transport is used
-                // because gRPC response values are always parsed correctly. Therefore
-                // the default $connectionType is set to 'grpc'
-                if (is_string($value)) {
```

Comment:
No, 

This was to compensate for the difference between the JSON and gRPC backends. We added this logic so that JSON's outputs matches that of gRPC.